keanu-python/keanu/optimization.py

```diff
@@ -0,0 +1,58 @@
+from py4j.java_gateway import java_import
+from keanu.base import JavaObjectWrapper
+from keanu.context import KeanuContext
+from keanu.inference import BayesNet
+from keanu.vertex import Vertex
+
+k = KeanuContext().jvm_view()
+
+java_import(k, "io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer")
+java_import(k, "io.improbable.keanu.algorithms.variational.optimizer.nongradient.NonGradientOptimizer")
+
+
+class Optimizer:
+    def __init__(self, optimizer):
+        self.optimizer = optimizer
+        self.net = optimizer.net
+
+    def max_a_posteriori(self):
+        return self.optimizer.maxAPosteriori()
+
+    def max_likelihood(self):
+        return self.optimizer.maxLikelihood()
+
+
+class GradientOptimizer(Optimizer):
+    def __init__(self, net, max_evaluations=None, relative_threshold=None, absolute_threshold=None):
+        builder = k.GradientOptimizer.builder()
+        if isinstance(net, BayesNet):
+            builder = builder.bayesianNetwork(net.unwrap())
```

Comment:
you don't need the `builder = ` here or on subsequent lines. Once you have a reference to the builder (line 27) you're good.